tests/transforms/test_batch_transform.py

```diff
@@ -0,0 +1,427 @@
+        return np.cos(weights[1] * np.cos(x)) + np.cos(weights[0] * np.sin(x))
+
+    def test_differentiable_autograd(self, diff_method):
+        """Test that a batch transform is differentiable when using
+        autograd"""
+        dev = qml.device("default.qubit", wires=2)
+        qnode = qml.QNode(self.circuit, dev, interface="autograd", diff_method=diff_method)
+
+        def cost(x, weights):
+            return self.my_transform(qnode, weights)(x)
+
+        weights = np.array([0.1, 0.2], requires_grad=True)
+        x = np.array(0.543, requires_grad=True)
+
+        res = cost(x, weights)
+        assert np.allclose(res, self.expval(x, weights))
+
+        grad = qml.grad(cost)(x, weights)
+        expected = qml.grad(self.expval)(x, weights)
+        assert all(np.allclose(g, e) for g, e in zip(grad, expected))
+
+    def test_differentiable_tf(self, diff_method):
+        """Test that a batch transform is differentiable when using
+        TensorFlow"""
+        if diff_method in ("parameter-shift", "finite-diff"):
+            pytest.skip("Does not support parameter-shift mode")
+
+        tf = pytest.importorskip("tensorflow")
+        dev = qml.device("default.qubit", wires=2)
+        qnode = qml.QNode(self.circuit, dev, interface="tf", diff_method=diff_method)
+
+        weights = tf.Variable([0.1, 0.2], dtype=tf.float64)
+        x = tf.Variable(0.543, dtype=tf.float64)
+
+        with tf.GradientTape() as tape:
+            res = self.my_transform(qnode, weights)(x)
+
+        assert np.allclose(res, self.expval(x, weights))
+
+        grad = tape.gradient(res, [x, weights])
+        expected = qml.grad(self.expval)(x.numpy(), weights.numpy())
+        assert all(np.allclose(g, e) for g, e in zip(grad, expected))
+
+    def test_differentiable_torch(self, diff_method):
+        """Test that a batch transform is differentiable when using
+        PyTorch"""
+        if diff_method == "backprop":
+            pytest.skip("Does not support backprop mode")
+
+        torch = pytest.importorskip("torch")
+        dev = qml.device("default.qubit", wires=2)
+        qnode = qml.QNode(self.circuit, dev, interface="torch", diff_method=diff_method)
+
+        weights = torch.tensor([0.1, 0.2], requires_grad=True)
+        x = torch.tensor(0.543, requires_grad=True)
+
+        res = self.my_transform(qnode, weights)(x)
+        expected = self.expval(x.detach().numpy(), weights.detach().numpy())
+        assert np.allclose(res.detach().numpy(), expected)
+
+        res.backward()
+        expected = qml.grad(self.expval)(x.detach().numpy(), weights.detach().numpy())
+        assert np.allclose(x.grad, expected[0])
+        assert np.allclose(weights.grad, expected[1])
+
+    def test_differentiable_jax(self, diff_method):
+        """Test that a batch transform is differentiable when using
+        jax"""
+        if diff_method in ("parameter-shift", "finite-diff"):

```

Comment:
Same here, I thought JAX supported parameter-shift?

Reply:
Still waiting on support 🙂 It will be coming in #1573